### PR TITLE
Add earler link to LSIF docs.

### PR DESCRIPTION
**Pull request by efritz**

### CODE REVIEW

File: doc/user/code_intelligence/basic_code_intelligence.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 Sourcegraph comes with built-in code intelligence provided by search-based heuristics.
 
+For precise code intelligence, see [LSIF](./lsif.md).
```
Comment (christinaforney) [inline doc/user/code_intelligence/basic_code_intelligence.md]:
```suggestion
If you are interested in enabling precise code intelligence on your repository, see our documentation on [LSIF](./lsif.md).
```

Comment (ryan-blunden) [inline doc/user/code_intelligence/basic_code_intelligence.md]:
```suggestion
Sourcegraph also supports precise code intelligence using the [Language Server Index Format (LSIF)](lsif.md).
```

Comment (efritz) [reply doc/user/code_intelligence/basic_code_intelligence.md]: *bump*, @christinaforney 's change made it in first.

Comment (ryan-blunden) [reply doc/user/code_intelligence/basic_code_intelligence.md]: I like hers so you're good to merge.

**Pull request merged**